test/new-e2e/tests/agent-metric-logs/linux-log-agent/file_tailing_utils_test.go

```diff
@@ -0,0 +1,109 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package logagent
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	fi "github.com/DataDog/datadog-agent/test/fakeintake/client"
+	componentos "github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/stretchr/testify/assert"
+)
+
+// generateLog generates and verifies log contents.
+func generateLog(s *LinuxVMFakeintakeSuite, content string) {
+	// Determine the OS and set the appropriate log path and command.
+	var logPath, cmd, checkCmd string
+	t := s.T()
+
+	// Get OS type using the OSType from vm_client.go from e2e/client package
+	osType := s.Env().VM.GetOSType()
+	var os string
+
+	switch osType {
```

Comment:
Instead of shelling out to append to the file you could always do something like:
```
	var logPath string
	switch osType {
	case "Windows":
		logPath = "C:\\logs\\hello-world.log"
	default: // Assuming Linux if not Windows.
		logPath = "/var/log/hello-world.log"
	}
	f, err := os.OpenFile(logPath,
		os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
	if err != nil {
		log.Println(err)
	}
	defer f.Close()
	if _, err := f.WriteString("text to append\n"); err != nil {
		log.Println(err)
	}
```

Reply:
Oh, never mind. Doesn't look like there is a way to open a file and write data to it on the VM (only copy files exists).

Reply:
We could add a feature to keep the session open, we can assign pipes in and/or out and write to the pipe instead of to the command:

https://pkg.go.dev/golang.org/x/crypto/ssh#Session.StdinPipe

Reply:
But I'm not sure if `ssh.Run/Shell` blocks. To be tested